src/panels/MissionsPanel/hooks.ts

```diff
@@ -39,3 +42,31 @@ export function useJumpToTime(fadeTime: number = 1): ReturnType {
     }
   };
 }
+
+export function useSelectedMission(): {
+  mission: Phase | undefined;
+  hasMission: boolean;
+} {
+  const { isInitialized, data, selectedMissionIdentifier } = useAppSelector(
+    (state) => state.missions
+  );
+
+  let mission: Phase | undefined = undefined;
+
+  // If a new mission is added, the missions order might have changed so we try to find
+  // the last one viewed
+  if (isInitialized && selectedMissionIdentifier) {
+    const selectedMission = data.missions.find(
```

Comment:
This speaks to using a map instead, IMO. I would rather add some complexity in the redux store than have to search the list of missions. 

And it would be nice to avoid having to use zero-indexing to get the missing if there is only one